module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,63 @@
+#include "compression.hpp"
+
+#include <algorithm>
+#include <iostream>
+
+pairVector compressGrayscale(const std::array<std::array<uint8_t, width>, height>& inputMap) {
+    pairVector outputVec;
+    outputVec.reserve(width * height);
+    uint8_t color = 0;
+    int value = 0;
+    int column = 0;
+
+    for (const auto& mapRow : inputMap) {
+        color = mapRow.front();
+        value = 0;
+        column = 0;
+
+        for (const auto& rowElement : mapRow) {
+            if (rowElement != color || column == width - 1) {
+                if (column == width - 1)
+                    value++;
+                outputVec.emplace_back(std::make_pair(color, (value)));
```

Comment:
Did you know that you can make pair without calling std::make_pair? Like this:
outputVec.emplace_back(color, value);
And why are you inserting (value) instead of value?

Reply:
Using (vale) instead of vector was made by mistake. 
I known that it is not necessary to use std::make_pair, but could you please explain, is it better or faster to use _outputVec.emplace_back(color, value)_ without std::make_pair?

Reply:
Emplace_back calls directly constructor with args. So when you use empace_back and use  std::make_pair You are doing it wrong, because you are creating temporary variable.
So You either  use make_pair and push_back OR directly call constructor using emplace_back with arguments.
There is discussion about this in the stackoverflow:
https://stackoverflow.com/questions/53900865/emplace-back-vs-push-back-when-inserting-a-pair-into-stdvector